net/WebSocketHandler.hpp

```diff
@@ -260,8 +269,9 @@ class WebSocketHandler : public ProtocolHandlerInterface
         }
 
         unsigned char *p = reinterpret_cast<unsigned char*>(&socket->getInBuffer()[0]);
-        const bool fin = p[0] & 0x80;
-        const WSOpCode code = static_cast<WSOpCode>(p[0] & 0x0f);
+        _lastFlags = p[0];
```

Comment:
What if the buffer is empty? Or if it can't be empty at this point, then assert that?

Reply:
A few lines up:
```
        const size_t len = socket->getInBuffer().size();
        if (len == 0)
            return false; // avoid logging.
```

Notice that I'm not changing logic; I'm only capturing the value (of the first byte) in a member.